tools/ci-ts/test-helpers/chainlinkClient.test.ts

```diff
@@ -0,0 +1,150 @@
+let containerMockState = {
+  Paused: false,
+}
+
+function setContainerMockState(newState: any) {
+  containerMockState = newState
+}
+
+const containerMock = {
+  pause: jest.fn(),
+  unpause: jest.fn(),
+  inspect: jest
+    .fn()
+    .mockImplementation(async () => ({ State: containerMockState })),
+}
+
+const dockerMock = {
+  getContainer: jest.fn().mockReturnValue(containerMock),
+}
+
+jest.mock('dockerode', () => jest.fn().mockReturnValue(dockerMock))
+jest.genMockFromModule('execa')
+jest.mock('execa')
+import Dockerode from 'dockerode'
```

Comment:
Does the test fail if we put the imports at the top of the file?

Reply:
it does, yeah because `jest.mock('dockerode', ...)` has to come before the dockerode import. And so the `dockerMock` has to be defined already too.

Reply:
I think jest should be hoisting it for us automatically now that I look at the docs: https://jestjs.io/docs/en/manual-mocks#using-with-es-module-imports